src/test/java/org/openrewrite/staticanalysis/EqualsAvoidsNullTest.java

```diff
@@ -473,4 +473,104 @@ public class A {
           )
         );
     }
+
+    @Issue("https://github.com/openrewrite/rewrite-static-analysis/issues/472")
+    @Nested
+    class equalsAvoidsNullNonIdempotent {
+
+        @Test
+        void raw() {
+            rewriteRun(
+              //language=java
+              java(
+                """
+                  public class Foo {
+                      public void bar() {
+                          "FOO".equals("BAR");
+                          "FOO".equalsIgnoreCase("BAR");
+                          "FOO".compareTo("BAR");
+                          "FOO".compareToIgnoreCase("BAR");
+                      }
+                  }
+                  """
+              )
+            );
+        }
+
+        /**
+         * ATM we cannot determine this case without breaking {@link ReplaceConstantMethodArg#one}
```

Comment:
@timtebeek i was not able to determine the states correctly. Maybe you can fix this case. Its possible to make the test work but not without breaking the pre existing once.

Reply:
Did you tried to solve it with the `DataFlow` APIs `ConstantFold` [feature](https://github.com/openrewrite/rewrite-analysis/blob/main/src/main/java/org/openrewrite/analysis/constantfold/ConstantFold.java#L41)? It determines the constant value if possible.